### PR TITLE
OSDOCS-3801: Add OVN-Kubernetes internal subnet configuration

### PR DESCRIPTION
**Pull request by jboxman-rh**
- https://issues.redhat.com/browse/OSDOCS-3801

Preview: https://50264--docspreview.netlify.app/openshift-enterprise/latest/networking/cluster-network-operator.html#nw-operator-cr-cno-object_cluster-network-operator

Version(s): 4.12

TODO: Squash commits

### CODE REVIEW

**Comment by jboxman-rh** [comment]
@pliurh, I've added descriptions for these new OVN-Kubernetes configuration fields. Can you confirm? Thanks!

I think we should include these as part of the OpenShift SDN to OVN-Kubernetes migration documentation as well?

File: modules/nw-operator-cr.adoc [hunk 1]
```diff
@@ -289,6 +289,20 @@ endif::ibm-cloud[]
 ====
  While migrating egress traffic, you can expect some disruption to workloads and service traffic until the Cluster Network Operator (CNO) successfully rolls out the changes.
 ====
+
+|`v4InternalSubnet`
+|
+If your existing network infrastructure overlaps with the `100.64.0.0/16` IPv4 subnet, you can specify a different IP address range for internal use by OVN-Kubernetes. You must ensure that the IP address range does not overlap with any other subnet used by your {product-title} installation. The IP address range must be large enough to allocate one IP address per cluster node.
```
Comment (pliurh) [inline modules/nw-operator-cr.adoc:295]: To be precise, the IP address range must be larger than the maximum number of nodes that can be added to the cluster.

For example, if the clusterNetwork CIDR is 10.128.0.0/14, and the hostPrefix is /23. The maximum number of nodes is 2^(23-14)=128. And as we need one IP for the gateway and one for broadcasting. So the internal IP range must be at least `/24`.

File: modules/nw-operator-cr.adoc [hunk 2]
```diff
@@ -289,6 +289,20 @@ endif::ibm-cloud[]
...
+
+This field cannot be changed after installation.
+|The default value is `100.64.0.0/16`.
+
+|`v6InternalSubnet`
+|
+If your existing network infrastructure overlaps with the `fd98::/48` IPv6 subnet, you can specify a different IP address range for internal use by OVN-Kubernetes. You must ensure that the IP address range does not overlap with any other subnet used by your {product-title} installation. The IP address range must be large enough to allocate one IP address per cluster node.
```
Comment (pliurh) [inline modules/nw-operator-cr.adoc:302]: ditto

**Comment by pliurh** [comment]
@weliang1 PTAL.

File: modules/nw-operator-cr.adoc (hunk 2 above)
Comment (jboxman-rh) [reply modules/nw-operator-cr.adoc:302]: @pliurh, I don't understand enough about IPv6 to be able to translate. It's probably not a `/24` though?

Comment (pliurh) [reply modules/nw-operator-cr.adoc]: The calculation is similar to the IPv4. If the clusterNetwork CIDR is fd01::/56, and the hostPrefix is /64. The maximum number of nodes is 2^(64-56)=256. And as we need one IP for the gateway, one for the network, and one for broadcasting. So the internal IP range must be at least /23.

File: modules/nw-operator-cr.adoc (hunk 1 above)
Comment (pliurh) [reply modules/nw-operator-cr.adoc]: There is an error in the calculation. The maximum number of nodes is 2^(23-14)=512, So the internal IP range must be at least /22.

File: modules/nw-operator-cr.adoc [hunk 3]
```diff
@@ -289,6 +289,22 @@ endif::ibm-cloud[]
 ====
  While migrating egress traffic, you can expect some disruption to workloads and service traffic until the Cluster Network Operator (CNO) successfully rolls out the changes.
 ====
+
+|`v4InternalSubnet`
+|
+If your existing network infrastructure overlaps with the `100.64.0.0/16` IPv4 subnet, you can specify a different IP address range for internal use by OVN-Kubernetes. You must ensure that the IP address range does not overlap with any other subnet used by your {product-title} installation. The IP address range must be larger than the maximium number of nodes that can be added to the cluster.
```
Comment (bmcelvee) [inline modules/nw-operator-cr.adoc:295]: s/maximium/maximum

There are two instances

**Comment by bmcelvee** [comment]
Just an edit and a reminder to squash your commits. Otherwise, LGTM!

**Comment by jboxman-rh** [comment]
@bmcelvee thanks!

**Pull request merged**